v2/cf-nodejs-logging-support/src/lib/logger/record-factory.ts

```diff
@@ -7,16 +7,25 @@ import ResponseAccessor from "../middleware/response-accessor";
 import ReqContext from "./context";
 
 export default class RecordFactory {
+    private static REDUCED_PLACEHOLDER = "redacted";
 
     // init a new record and assign fields with output "msg-log"
     static buildMsgRecord(args: Array<any>, context?: ReqContext): any {
 
-        const msgLogFields = Config.getInstance().getMsgFields();
+        const configInstance = Config.getInstance();
+        const msgLogFields = configInstance.getMsgFields();
         let record: any = {
             "level": "info",
         };
 
         msgLogFields.forEach(field => {
+            if (configInstance.isRedactedField(field.name)) {
+                record[field.name] = this.REDUCED_PLACEHOLDER;
```

Comment:
For fields being redacted following behavior is expected
* If the value of the field is present => set to "redacted"
* If the value is not present => omit field